pgxn/neon/unstable_extensions.c

```diff
@@ -0,0 +1,183 @@
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "postgres.h"
+
+#include "nodes/plannodes.h"
+#include "nodes/parsenodes.h"
+#include "tcop/utility.h"
+#include "utils/errcodes.h"
+#include "utils/guc.h"
+
+#include "unstable_extensions.h"
+
+#if PG_MAJORVERSION_NUM < 16
+#define guc_free(ptr) free(ptr)
+#define guc_malloc(level, size) malloc(size)
+#define guc_strdup(level, s) strdup(s)
+#endif
+
+typedef struct UnstableExtensions
+{
+	size_t	extc;
+	char   *extv[FLEXIBLE_ARRAY_MEMBER];
+} UnstableExtensions;
+
+static bool					allow_unstable_extensions = false;
+static char				   *unstable_extensions_str = NULL;
+static UnstableExtensions  *unstable_extensions = NULL;
+
+static ProcessUtility_hook_type PreviousProcessUtilityHook = NULL;
+
+static bool
+check_unstable_extensions(char **newval, void **extra, GucSource source)
+{
+	char			   *n;
+	UnstableExtensions *exts;
+	char			   *curr_ext;
+	size_t				max_extc;
+
+	if (*newval == NULL || (*newval)[0] == '\0')
+		return true;
+
+	/* At this point, we know we have at least 1 extension like: ext */
+	max_extc = 1;
+	for (size_t i = 0; i < strlen(*newval); ++i)
+	{
+		if ((*newval)[i] == ',')
+			++max_extc;
+	}
+
+	exts = guc_malloc(ERROR, sizeof(*exts) + max_extc * sizeof(char *));
```

Comment:
cc @knizhnik to review the C strings trickery :) Not sure I can do it quickly today

Reply:
There is no trickery, just usual flexible array type stuff at the end of the struct fwiw.